### ISSUE TITLE
sleeping in ConditionalRun inhibits SIGINT

### ISSUE THREAD

**Issue by ktf** [FairRootGroup/FairMQ#82]
I've a device which overrides `ConditionalRun()` and sleeps for a long time. Looks like any external SIGINT / SIGTERM sent to it is delayed until the end of the sleep. Is that expected? How can I have some interruptible sleep for a given device?

**Comment by rbx** [comment]
We catch the signals and let the device finish gracefully. Any Send/Receive calls will return with `-2` and device state will be changed to something other than RUNNING. Indeed your sleep will continue uninterrupted in this case. 

What we could do is to offer something like  `FairMQDevice::WaitFor(chrono::duration)` method that internally will wait on interruption condition variable for the given duration. So for you as a user it will behave just like sleep, but it will be still interruptible by signals and state changes.

Would this help you?

**Comment by dennisklein** [comment]
The new API

```cpp
template<class Rep, class Period>
bool FairMQDevice::WaitFor(std::chrono::duration<Rep, Period> const& duration);
```

is now available for testing in FairMQ `v1.3.3`.

**Comment by dennisklein** [comment]
Thx!
